### PR TITLE
modem: cmux: added validation of cmux frame length

### LINKED ISSUE #81053: modem: modem_cmux: Losing bytes in a cmux frame can cause long resync
**Is your enhancement proposal related to a problem? Please describe.**
If the length byte(s) in a CMUX frame header is lost, the state machine decoding the frame may in some cases wait for an invalid number of bytes before it resyncs to a new frame, which can be observed as non-responsive modem communication

**Describe the solution you'd like**
The proposed solution is to verify that the frame header length is within the receive buffer parameters, if not - drop the frame and wait for a new to resync

**Describe alternatives you've considered**
None

**Additional context**
A PR is submitted: #80663

### PR DESCRIPTION
**Pull request by henr1k**
Added validation of frame length to avoid 
waiting for an invalid number of bytes to be received.
If the frame length is longer than expected, the frame is dropped.

Signed-off-by: Henrik Skreslet [henrik.skreslet@gmail.com](mailto:henrik.skreslet@gmail.com)

### CODE REVIEW

**Comment by henr1k** [comment]
"Clone" of https://github.com/zephyrproject-rtos/zephyr/pull/80505

File: subsys/modem/modem_cmux.c [hunk 1]
```diff
@@ -816,6 +816,14 @@ static void modem_cmux_process_received_byte(struct modem_cmux *cmux, uint8_t by
 		/* Get last 8 bits of data length */
 		cmux->frame.data_len |= ((uint16_t)byte) << 7;
 
+		if (cmux->frame.data_len > cmux->receive_buf_size) {
+			LOG_ERR("Indicated frame data length %u exceeds receive buffer size %u",
```
Comment (tomi-font) [inline subsys/modem/modem_cmux.c:820]: Why `Indicated`? Wouldn't it rather be `Received`?

**Review by tomi-font — CHANGES_REQUESTED** [review]
Looks good, but you'll need to address the compliance errors shown [here](https://github.com/zephyrproject-rtos/zephyr/actions/runs/11608112986/job/32325661329?pr=80663#step:11:47) (lines 46 to 54).
When you make those changes, make sure to amend your commit (`git commit --amend`) and then force push your branch (`git push -f`).

File: subsys/modem/modem_cmux.c (hunk 1 above)
Comment (henr1k) [reply subsys/modem/modem_cmux.c:820]: I guess one could also say `Invalid` as well?

**Comment by henr1k** [comment]
> Looks good, but you'll need to address the compliance errors shown [here](https://github.com/zephyrproject-rtos/zephyr/actions/runs/11608112986/job/32325661329?pr=80663#step:11:47) (lines 46 to 54). When you make those changes, make sure to amend your commit (`git commit --amend`) and then force push your branch (`git push -f`).

I'm trying to run check_compliance.py locally, but I'm not getting any errors, what am I doing wrong? ;D

`(zephyr_venv) ➜  zephyr git:(pr_add_cmux_length_validation) ./scripts/ci/check_compliance.py -c pr_add_cmux_length_validation..`

File: subsys/modem/modem_cmux.c (hunk 1 above)
Comment (tomi-font) [reply subsys/modem/modem_cmux.c:820]: Ah, now I think I see what you mean with `Indicated`. Well, whatever, this was a minor remark anyway.

**Comment by tomi-font** [comment]
> I'm trying to run check_compliance.py locally, but I'm not getting any errors, what am I doing wrong? ;D

I don't know! But the errors seem easy to fix just based on the CI output, at least to me.

File: tests/subsys/modem/modem_cmux/src/main.c [hunk 2]
```diff
@@ -148,6 +148,14 @@ static uint8_t cmux_frame_dlci2_at_newline[] = {0xF9, 0x0B, 0xEF, 0x05, 0x0D, 0x
 
 static uint8_t cmux_frame_data_dlci2_at_newline[] = {0x0D, 0x0A};
 
+/*************************************************************************************************/
+/*                                     DLCI2 AT CMUX error frames								 */
```
Comment (tomi-font) [inline tests/subsys/modem/modem_cmux/src/main.c:152]: The closing `*/` on this line is still way too far. It should be aligned as the rest.

Comment (henr1k) [reply tests/subsys/modem/modem_cmux/src/main.c]: I think I'm being fooled by vscode, should be better now. :)

**Review by tomi-font — DISMISSED** [review]
:crossed_fingers:

**Review by bjarki-andreasen — DISMISSED** [review]

**Review by tomi-font — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Comment by henr1k** [comment]
Anything else I should do before this is merged? :)

**Comment by mmahadevan108** [comment]
@henr1k , can you add Zephyr issue detailing the issue that is fixed and link it to this PR.

**Comment by henr1k** [comment]
#81053

**Pull request merged**